tests/Propel/Tests/Generator/Builder/Om/GeneratedObjectDateColumnTypeTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * MIT License. This file is part of the Propel package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Propel\Tests\Generator\Builder\Om;
+
+use GeneratedObjectDateColumnTypeEntity;
+
+use DateTimeImmutable;
+use PDO;
+use PHPUnit\Framework\Exception;
+use PHPUnit\Framework\InvalidArgumentException;
+use PHPUnit\Framework\MockObject\ClassAlreadyExistsException;
+use PHPUnit\Framework\MockObject\ClassIsFinalException;
+use PHPUnit\Framework\MockObject\DuplicateMethodException;
+use PHPUnit\Framework\MockObject\InvalidMethodNameException;
+use PHPUnit\Framework\MockObject\OriginalConstructorInvocationRequiredException;
+use PHPUnit\Framework\MockObject\ReflectionException;
+use PHPUnit\Framework\MockObject\RuntimeException;
+use PHPUnit\Framework\MockObject\UnknownTypeException;
+use PHPUnit\Framework\MockObject\IncompatibleReturnValueException;
+use Propel\Generator\Util\QuickBuilder;
+use Propel\Runtime\Connection\ConnectionInterface;
+use Propel\Runtime\Connection\PdoConnection;
+use Propel\Runtime\Connection\StatementInterface;
+use Propel\Runtime\Connection\StatementWrapper;
+use Propel\Tests\TestCase;
+
+class GeneratedObjectDateColumnTypeTest extends TestCase
+{
+    public function setUp(): void
+    {
+        if (!\class_exists('GeneratedObjectDateColumnTypeEntity')) {
+            $schema = <<<'XML'
+<database name="generated_object_date_column_type">
+    <table name="generated_object_date_column_type_entity">
+        <column name="id" primaryKey="true" type="INTEGER" autoIncrement="true"/>
+        <column name="datecolumn" type="DATE"/>
+    </table>
+</database>
+XML;
+            QuickBuilder::buildSchema($schema);
+        }
+    }
+
+    public function testInsertDateColumn(): void
+    {
+        assert(class_exists(GeneratedObjectDateColumnTypeEntity::class));
+        $e = new GeneratedObjectDateColumnTypeEntity();
+        $this->assertTrue(method_exists($e, 'setDatecolumn'));
```

Comment:
You sometimes prefix functions with a `\` and sometimes not. @dereuromark do we have a CS fixer to align those?

Reply:
So far we are not using this yet. If we would, any fixer should normalize this then later.

Reply:
But as long as there is no sniffer for it to fix it in either direction, I would not bother about this now.

Reply:
@stereomon I processed your feedback, could you greenlight this?

Also: you might be interested in https://github.com/slevomat/coding-standard#slevomatcodingstandardnamespacesfullyqualifiedglobalfunctions- for the global-namespaced functions :smile:

Reply:
@nederdirk Yeah, we are so far not doing this as this is somewhat a nano-opt (small benefit) while having a lot of more noise in the use statements.
So far, we are using normal function usage (that also does allow overwriting for exception cases when needed).